adapters/lora-vertx-base/src/main/java/org/eclipse/hono/adapter/lora/providers/TheThingsStackProvider.java

```diff
@@ -179,4 +186,14 @@ protected LoraMetaData getMetaData(final JsonObject loraMessage) {
 
         return data;
     }
+
+    @Override
+    protected JsonObject getCommandPayload(final Buffer payload, final String deviceId, final String subject) {
+        final JsonObject json = new JsonObject();
+        json.put(FIELD_THE_THINGS_STACK_DECODED_PAYLOAD, BaseEncoding.base16().encode(payload.getBytes()));
+        json.put(FIELD_THE_THINGS_STACK_CONFIRMED, false);
+        json.put(FIELD_THE_THINGS_STACK_PRIORITY, VALUE_THE_THINGS_STACK_NORMAL);
+        json.put(FIELD_THE_THINGS_STACK_PORT, Integer.parseInt(subject));
```

Comment:
this will throw a ParseException if the subject is not the string representation of an integer. IMHO you should enclose this in a corresponding try/catch block and ignore the subject if it is not an integer.

How about adding similar logic to the other providers that support sending commands to devices?

Reply:
changed it and added it to other providers.

Reply:
changed it and added it to other providers.